### PR TITLE
Switch from puma to passenger

### PR DESCRIPTION
**Pull request by vuntz**
Doing some preliminary tests with passenger. So far, I'm not able to reproduce the SQLite3::BusyException exception...

This patch is not fully tested (I configured things manually for my testing).

We should still discuss whether the switch makes sense, of course.

### CODE REVIEW

**Comment by tboerger** [comment]
And who will maintain this monster? Maintaining passenger is a pain in the ass. Beside that we need to drop it with cloud 7 again when we introduce websockets as they are unsupported by the free passenger version. So I'm really not convinced...

**Comment by vuntz** [comment]
> And who will maintain this monster?

It's already maintained internally, so it comes for free.

**Comment by vuntz** [comment]
> Beside that we need to drop it with cloud 7 again when we introduce websockets as they are unsupported by the free passenger version.

"WebSockets work great on both the open source variant of Phusion Passenger, as well as on Phusion Passenger Enterprise. "

https://github.com/phusion/passenger-ruby-websocket-demo

**Comment by vuntz** [comment]
And again, this needs to be discussed, as mentioned in the PR. I only opened the PR to allow people to test things.

**Comment by rsalevsky** [comment]
I have the same experience with passenger as @tboerger. I have two projects running on passenger and both makes just problems. One is terrible slow while parallel requests and the other one throws random error messages after some time.

**Comment by tboerger** [comment]
Ok, so they changed the websocket thing. But I still have bad experience related to performance and stability like @rsalevsky have

File: crowbar_framework/lib/crowbar/logger/formatter.rb [hunk 1]
```diff
@@ -24,7 +24,7 @@ class Formatter < ::Logger::Formatter
       def call(severity, time, progname, msg)
         threads = ENV["CROWBAR_THREADS"]
         # if env var is not set, default is multi-threaded
-        if threads.to_s.empty? || threads.to_i > 1
+        if !threads.to_s.empty? || threads.to_i > 1
```
Comment (tboerger) [inline crowbar_framework/lib/crowbar/logger/formatter.rb]: `if threads.to_s.present? || threads.to_i > 1`

**Comment by tboerger** [comment]
A general question, why do you stick with a standalone passenger instead of the apache module?

**Comment by vuntz** [comment]
> A general question, why do you stick with a standalone passenger instead of the apache module?

It's mentioned in the commit message: because we have cookbooks that can reload or restart apache, using the module is actually pretty bad as the cookbooks can kill the rails app. And this actually happens during the installer... So we can't do that.

**Comment by vuntz** [comment]
This was an experiment; unless we decide to really do it, let's close this.

**Pull request closed without merging**